### PR TITLE
Remove duplicate and random algorithms from tests

### PR DESCRIPTION
**Pull request by Ornias1993**
This is a spinoff from #8941 seperating general, unrelated, test changes from the zstd PR.

TL:DR: 
- Random tests (`send-c_verify_ratio`) lead to inpredicatble results
- Duplicate algorithms lead to duplicate execution of tests
- Combined the effects are even worse and lead to false positives

Signed-off-by: Kjeld Schouten-Lebbing <kjeld@schouten-lebbing.nl>

### Motivation and Context
During my work on investigating the compression test suite for #8941 a few things became appearant:
`send-c_verify_ratio` used randomly selected compression algorithms.  Those algorithms differ widely in performance and for the majority consist of different levels of gzip. This lead to difficulty comparing performance of tests over multiple runs. This also lead to situations where tests ended with a false-positive (pass) result, due to algorithms being skipped.

During my research into this there seemed to be 2 versions of multi-compression test suits:
1. Randomly selected from a pool listing all different levels of algorithms
2. Fixed selective non-random selection of algorithms

The 2 tests using version 1, both suffered from the same problems. (listed above) This wouldn't be that bad, but it gets worse when new algorithms get added with multiple levels.

### Description
This change removes duplicate algorithms from all compression tests (and leaves just 1 of each kind), including the random compression pool.

It also removed the random draw of algorithms from `send-c_verify_ratio`, to make sure results are repeatable. This also removes one whole loop of the test and thus improves performance.

### Is * taken into account?
The following is taken into account:
- There is a seperate test for testing aliasses, those should work regardless
- If many more algorithms get added, possibly in the future in some places random selection might be needed, but still should only list one level each to prevent limiting the chances of drawing a non-level based algorithm like lz4
- If seperate levels work totally differently, a seperate test for that algorithm would be a cleaner solution than adding all levels everywhere.

### How Has This Been Tested?
i've been debating and testing multiple versions/levels of these changes for about a week now. After I came up with them due to (unrelated) issues in  #8941

### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Performance enhancement (non-breaking change which improves efficiency)
- [X] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
- [X] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [X] I have updated the documentation accordingly.
- [X] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [X] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [] All new and existing tests passed.
- [X] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by Ornias1993** [comment]
*note
Small thank-you to @c0d3z3r0 for mentioning general changes should be kept out of zstd when possible. If I complain about those of others, my own should be seperated as well :)

**Comment by Ornias1993** [comment]
@behlendorf You might want a restart on those failed tests (unrelated).
But I don't think its worth it, it has absolutely nothing to do with the PR and would've been caught by those tests regardless if there where any issues with this.

**Comment by c0d3z3r0** [comment]
What about `tests/zfs-tests/include/libtest.shlib`?

File: tests/zfs-tests/tests/functional/rsend/rsend_012_pos.ksh [hunk 1]
```diff
@@ -138,9 +138,7 @@ done
 
 for vol in "$POOL/vol" "$POOL/$FS/vol" ; do
 	rand_set_prop $vol checksum "on" "off" "fletcher2" "fletcher4" "sha256"
-	rand_set_prop $vol compression "on" "off" "lzjb" "gzip" \
-		"gzip-1" "gzip-2" "gzip-3" "gzip-4" "gzip-5" "gzip-6"   \
-		"gzip-7" "gzip-8" "gzip-9"
+	rand_set_prop $vol compression "off" "lzjb" "gzip" "lz4"
```
Comment (c0d3z3r0) [inline tests/zfs-tests/tests/functional/rsend/rsend_012_pos.ksh]: add zle? (separate PR?)

**Comment by c0d3z3r0** [comment]
Oh, I just stumbled over this... aren't we missing zle and lz4 in `function get_compress_opts` in `tests/zfs-tests/include/libtest.shlib`

**Comment by Ornias1993** [comment]
@c0d3z3r0 I purposefully left this PR to do one thing only:
Remove duplicate algorithms and random selected algorithms where possible.

I have decided not to add the added algorithms brainslayer added to my PR.

**Comment by Ornias1993** [comment]
@c0d3z3r0 
- lz4 is included in `libtest.shlib` as `on`
- I've decided not to remove the gzip list there, I am not 100% certain that list should contain just 1 gzip variant.... The way brainslayer did those changes didn;t look completely right to me, throwing in `GZIP_OPTS` with just 1 value and adding `ZSTD_OPTS` with also just one value...

File: tests/zfs-tests/tests/functional/rsend/rsend_012_pos.ksh (hunk 1 above)
Comment (Ornias1993) [reply tests/zfs-tests/tests/functional/rsend/rsend_012_pos.ksh]: ZLE with random writen data (which is often used) is basically the same as `compression=off`

**Comment by Ornias1993** [comment]
@c0d3z3r0 
You made me realise the fact we might need a bigger refactor of this, I'll close this for now while I work on that. We can always reopen if i'm not happy with how that works out :)
